examples/i3-cycle-focus.py

```diff
@@ -7,15 +12,19 @@
 from argparse import ArgumentParser
 import i3ipc
 
-SOCKET_FILE = '/tmp/i3-cycle-focus'
+SOCKET_FILE = '/tmp/.i3-cycle-focus.sock'
 MAX_WIN_HISTORY = 16
 UPDATE_DELAY = 2.0
 
 
+def on_shutdown(i3_conn, e):
+    os._exit(0)
```

Comment:
Why are you using `_exit`?

Reply:
sys.exit() leaves one of the threads hanging around.